File: Comentario.java

public class Comentario{

    private String texto;
    private Comentario resposta;
    private int curtidas;
    //private Usuario user; //*** fazer */
    
    public Comentario(String texto){
        this.texto = texto;
        this.resposta = null;
        //this.user = null; //comentario anonimo(???)
    }

    //necessario implementacao de usuario
    /*
    public Comentario(String texto,Usuario user){
        this.texto = texto;
        this.resposta = null;
        //this.user = user; 
    }
    */

    public void addResposta(Comentario c){
        this.resposta = c;
    }

    public void addResposta(String texto){
        this.texto = texto;
    }

    public void removeComentario(){
        
    }

    public void like(){
        this.curtidas++;
    }

    public void dislike(){
        this.curtidas--;
    }

    public String getTexto(){
        return this.texto;
    }

    /*
    public getUsuario(){
        return user;
    }
    */
}

File: Comentario.dart

class Comentario{

    static var _texto;
    var _resposta;
    int _curtidas = 0;
    //private Usuario user; //*** fazer */
    
    Comentario(texto){
        _texto = texto;
        _resposta = null; 
        _curtidas = 0; 
        //this.user = null; //comentario anonimo(???)
    }

    //necessario implementacao de usuario
    /*
    public Comentario(String texto,Usuario user){
        this.texto = texto;
        this.resposta = null;
        //this.user = user; 
    }
    */

    void addResposta(texto){
        _texto = texto;
    }

    void removeComentario(){
        
    }

    void like(){
        _curtidas++;
    }

    void dislike(){
        _curtidas--;
    }

    String getTexto(){
        return _texto;
    }

    /*
    public getUsuario(){
        return user;
    }
    */
}